src/ODEmodel.jl

```diff
@@ -62,16 +62,17 @@ function predict(p, g_0::Real, t, nG1::Integer, nG2::Integer, nD1, nD2, retVec =
         D2 = zeros(nD2)
     end
 
-    v = [ones(nG1) * p[5] * g_0 / nG1; ones(nG2) * (1.0 - p[5]) * g_0 / nG2; D1; D2]
     A = ODEjac(p, nG1, nG2, nD1, nD2)
 
     if t isa Real
-        v = ExponentialUtilities.expv(t, A, v)
-        G1 = sum(v[1:nG1]) + sum(v[(nG1 + nG2 + 1):(nG1 + nG2 + nD1)])
-        G2 = sum(v[(nG1 + 1):(nG1 + nG2)]) + sum(v[(nG1 + nG2 + nD1 + 1):(nG1 + nG2 + nD1 + nD2)])
+        G1 = sum(vec[1:nG1]) + sum(vec[(nG1 + nG2 + 1):(nG1 + nG2 + nD1)])
+        G2 = sum(vec[(nG1 + 1):(nG1 + nG2)]) + sum(vec[(nG1 + nG2 + nD1 + 1):(nG1 + nG2 + nD1 + nD2)])
+        newV = expv(t, A, vec)
+        return G1, G2, newV
     else
         # Some assumptions
         @assert t[1] == 0.0
+        v = [ones(nG1) * p[5] * g_0 / nG1; ones(nG2) * (1.0 - p[5]) * g_0 / nG2; D1; D2]
         rmul!(A, t[2])
         A = LinearAlgebra.exp!(A)
 
```

Comment:
I have been struggling with this for quite a few hours. For some parameter sets, the state vector values go to almost infinity. Do you see anything wrong with these changes @aarmey?
I inputted the `v` vector to the `predict`. I moved the `expv(...)` to the bottom because the state vector it receives is actually `v = A * v`.

Reply:
The time we give the `expv()` has to be `dt` or just `t`?

Reply:
I tried both, didn't change the infinity result, though.

Reply:
The way you've changed the function, you have to be careful to set `vec` to something if `t` is a time value. How about have `vec = nothing`, then if `vec is nothing` set it equal to the default value for `v`?

Reply:
After this, could you provide a set of function arguments that cause this to be infinite? This inside a test would be very helpful.

Reply:
I will try it right now.